### DISCUSSION TITLE
What goes in /predicates?

### DISCUSSION THREAD

**Discussion by cbizon** [NCATSTranslator/TranslatorArchitecture#31] [General]
Suppose we stipulate that KPs will perform category and predicate deductions.   So if a KP contains, from its source an edge like
ChemicalSubstance-inhibits->Gene,
It will return that edge if a TRAPI query comes in that asks for ChemicalSubstance-related_to->NamedThing.

Now, what should go in the /predicates endpoint?

I can imagine two possibilities:

1) The most specific form of the generic edge goes into /predicates.  So in this case, we would just have
```
biolink:ChemicalSubstance {
   biolink:Gene: [biolink:inhibits]
}
```

2) All forms of the edge that could be asked for:
```
biolink:ChemicalSubstance{
   biolink:Gene: [biolink:inhibits, biolink:affects, ..., biolink:related_to]
   biolink:GeneticEntity: [biolink:inhibits,..., biolink:related_to]
   ...
   biolink:NamedThing: [biolink:related_to]
}
...
biolink:NamedThing {
  biolink:Gene: [...]
  ...
  biolink:NamedThing: [biolink:related_to]
}
```

On today's call I believe @vdancik was advocating for a third option, but I'm still a bit unclear what that was...

Between these two, I can see arguments for each, but I prefer the first.

Thoughts?

**Comment by edeutsch** [comment]
I am thinking that KPs should advertise (via /predicates) the exact predicates that they encode. So if a KP understands from some other source that a ChemicalSubstance - inhibits - Protein
then they should advertise exactly that.
If some other actor wants to infer that "inhibits" implies "related_to" and that a protein is translated from a gene, that's fine. but not the job of the KP?

**Reply by cbizon** [reply]
>If some other actor wants to infer that "inhibits" implies "related_to" and that a protein is translated from a gene, that's fine. but not the job of the KP?

I think we need to break that down a bit.    There is the part about replacing protein with gene, and I agree that's out of scope for the current discussion.

So for this example, let's just consider the predicate part: if the KP knows that ChemicalSubstance-[inhibits]->Protein, and it advertises just that, then that's version 1 above.  But if that KP receives a query like ChemicalSubstance-[related_to]->Protein, it will return the inhibits information, right?

**Reply by edeutsch** [reply]
Well this is clearly a crucial decision that we need to make. Perhaps to play the role of the opposition in the debate, I rather think you're asking KPs to do reasoning here, which was not intended. I'm going to say that in the above scenario, the KP doesn't know "ChemicalSubstance-[related_to]->Protein". All the KP knows is that "ChemicalSubstance-[inhibits]->Protein". If you ask the KP "ChemicalSubstance-[related_to]->Protein?" you will get no answer. The job of a reasoning agent is to ask the KP "ChemicalSubstance-[]->Protein?" and sift through the results and reason that if the KP says "ChemicalSubstance-[inhibits]->Protein", then on can reason that "ChemicalSubstance-[related_to]->Protein". That's what reasoners do. Should KPs be reasoners? I don't feel that strongly either way myself, but I strongly advocate that we debate this out and come up with a clear Translator policy for everyone to follow!

**Reply by cbizon** [reply]
Yes, that's right.  I think there is some very basic reasoning that KPs should engage in.  I won't reproduce my reasons / scope  here, but they are in https://github.com/NCATSTranslator/TranslatorArchitecture/discussions/25.   And the related PR here: https://github.com/NCATSTranslator/TranslatorArchitecture/pull/28

**Reply by edeutsch** [reply]
Is there a reason to think that ARA reasoners might be less successful in their own reasoning if KPs are providing information that is not "primary" assertions, but rather assertions that are already partly reasoned in a way that the ARA does not easily know?

**Reply by cbizon** [reply]
> assertions that are already partly reasoned in a way that the ARA does not easily know?

I think that this is a key point: the kinds of reasoning we are talking about are ones that the ARA can easily know, and would probably(possibly?) even be expected by a 'naive' ARA.
